### PR TITLE
[BEAM-8676] sdks/java: gax and grpc upgrades

### PR DESCRIPTION
**Pull request by suztomo**
https://issues.apache.org/jira/browse/BEAM-8676. This PR is a revival of https://github.com/apache/beam/pull/10438.

Gax and gRPC work together. Upgrading their dependencies to higher versions.

- The latest gax-grpc:1.52.0 depends on gRPC 1.25.0. (So not setting grpc 1.26.0)
- Although grpc 1.25 declares Guava 28.1 dependency, grpc 1.25 does not have linkage errors with Guava 25.1-jre (which current master has).

# Which modules use gax and grpc?
`find . -name 'build.gradle' |xargs grep -ir library.java.gax` (and grpc) gave

- ./sdks/java/io/google-cloud-platform/build.gradle
- ./sdks/java/extensions/sql/datacatalog/build.gradle
- ./sdks/java/extensions/sql/zetasql/build.gradle

# Linkage Check

## beam-sdks-java-io-google-cloud-platform
- before: https://gist.github.com/suztomo/d0446fb63aea142db1a8086da702a04d (master ba4dc72c)
- after: https://gist.github.com/suztomo/5c0cd8699f019d03ae8bc0a58e3c7116 (this PR b4bd0bc5)

This PR removes the following linkage errors:

```
suztomo@suxtomo24:~/beam6$ diff /tmp/linkagecheck-beam-sdks-java-io-google-cloud-platform-ba4dc72c /tmp/linkagecheck-beam-sdks-java-io-google-cloud-platform-b4bd0bc5 
22,36d21
< Class io.opencensus.trace.Tracestate is not found;
<   referenced by 1 class file
<     io.opencensus.contrib.http.util.CloudTraceFormat (opencensus-contrib-http-util-0.18.0.jar)
< Class io.opencensus.trace.Tracestate$Builder is not found;
<   referenced by 1 class file
<     io.opencensus.contrib.http.util.CloudTraceFormat (opencensus-contrib-http-util-0.18.0.jar)
< Class com.google.common.truth.Truth is not found;
<   referenced by 1 class file
<     io.grpc.internal.testing.AbstractTransportTest (grpc-testing-1.17.1.jar)
< Class com.google.common.truth.StringSubject is not found;
<   referenced by 1 class file
<     io.grpc.internal.testing.AbstractTransportTest (grpc-testing-1.17.1.jar)
< Class com.google.common.truth.LongSubject is not found;
<   referenced by 1 class file
<     io.grpc.internal.testing.AbstractTransportTest (grpc-testing-1.17.1.jar)
44,55d28
< Class org.apache.tools.ant.Task is not found;
<   referenced by 1 class file
<     org.mockito.cglib.transform.AbstractProcessTask (mockito-core-1.9.5.jar)
< Class org.apache.tools.ant.types.FileSet is not found;
<   referenced by 1 class file
<     org.mockito.cglib.transform.AbstractProcessTask (mockito-core-1.9.5.jar)
< Class org.apache.tools.ant.DirectoryScanner is not found;
<   referenced by 1 class file
<     org.mockito.cglib.transform.AbstractProcessTask (mockito-core-1.9.5.jar)
< Class org.apache.tools.ant.BuildException is not found;
<   referenced by 1 class file
<     org.mockito.cglib.transform.AbstractProcessTask (mockito-core-1.9.5.jar)
870,872d842
< (opencensus-api-0.15.0.jar) io.opencensus.trace.SpanContext's method create(io.opencensus.trace.TraceId arg1, io.opencensus.trace.SpanId arg2, io.opencensus.trace.TraceOptions arg3, io.opencensus.trace.Tracestate arg4) is not found;
<   referenced by 1 class file
<     io.opencensus.contrib.http.util.CloudTraceFormat (opencensus-contrib-http-util-0.18.0.jar)
883c853
< BUILD SUCCESSFUL in 31s
---
> BUILD SUCCESSFUL in 30s

```

## beam-sdks-java-extensions-sql-zetasql

Linkage Checker does not work for beam-sdks-java-extensions-sql-zetasql. (Bug: GoogleCloudPlatform/cloud-opensource-java#1087)

## beam-sdks-java-extensions-sql-datacatalog

Linkage Checker does not work for beam-sdks-java-extensions-sql-datacatalog https://gist.github.com/suztomo/9c7723a2d126459bebfa2f52b5573757  (the same reason as above) 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: sdks/java/io/google-cloud-platform/build.gradle [hunk 1]
```diff
@@ -50,6 +51,7 @@ dependencies {
   compile library.java.grpc_all
   compile library.java.grpc_auth
   compile library.java.grpc_core
+  compile library.java.grpc_context
```
Comment (suztomo) [inline sdks/java/io/google-cloud-platform/build.gradle]: Without declaring grpc-context, its old version sneaks in the class path. Error from linkage checker:

```
(grpc-context-1.16.1.jar) io.grpc.Deadline's method getSystemTicker() is not found;
  referenced by 1 class file
    io.grpc.internal.AbstractServerImplBuilder (grpc-core-1.25.0.jar)
```
https://gist.github.com/suztomo/9af333b78b157d6a829d59ac5353e01d#file-gistfile1-txt-L856

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 2]
```diff
@@ -372,7 +373,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def google_cloud_bigdataoss_version = "1.9.16"
     def google_cloud_core_version = "1.61.0"
     def google_cloud_spanner_version = "1.6.0"
-    def grpc_version = "1.17.1"
+    def grpc_version = "1.25.0"
```
Comment (suztomo) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: The latest version of gRPC is 1.26.0, but setting 1.25.0 as gax-grpc:1.52.0 depends on 1.25.0.

File: sdks/java/io/google-cloud-platform/build.gradle [hunk 3]
```diff
@@ -33,6 +33,7 @@ dependencies {
   compile project(":sdks:java:extensions:protobuf")
   compile library.java.avro
   compile library.java.bigdataoss_util
+  compile library.java.gax
```
Comment (suztomo) [inline sdks/java/io/google-cloud-platform/build.gradle]: As BigQueryServicesImpl uses gax's `com.google.api.gax.core.FixedCredentialsProvider` (undeclared-but-used dependency), declaring it explicitly.

**Comment by suztomo** [comment]
R: @lukecwik, @kennknowles

**Comment by apilloud** [comment]
Run Java PostCommit

**Comment by apilloud** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by apilloud** [comment]
Run BigQueryIO Streaming Performance Test Java

**Comment by apilloud** [comment]
Run Dataflow ValidatesRunner

**Comment by apilloud** [comment]
Run Spark ValidatesRunner

**Comment by apilloud** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
It seems Jenkins went down while running the 4 checks.

java.io.IOException: Backing channel 'JNLP4-connect connection from 255.248.197.104.bc.googleusercontent.com/104.197.248.255:33122' is disconnected.

**Comment by yifanzou** [comment]
Run Java PostCommit

**Comment by yifanzou** [comment]
Run Java HadoopFormatIO Performance Test

**Comment by yifanzou** [comment]
Run BigQueryIO Streaming Performance Test Java

**Comment by yifanzou** [comment]
Run Spark ValidatesRunner

**Comment by yifanzou** [comment]
Run SQL Postcommit

**Comment by suztomo** [comment]
Apache Spark Runner ValidatesRunner Tests failed:
java.lang.IllegalStateException: No TransformEvaluator registered for UNBOUNDED transform TestStream

Google Cloud Dataflow Runner ValidatesRunner Tests failed:
java.lang.RuntimeException: Workflow failed. Causes: Unknown streaming source: test_stream


I will check whether they’re caused by this dependency upgrades or not.

**Comment by suztomo** [comment]
@yifanzou thanks

**Comment by suztomo** [comment]
The two tests failed because of this [BEAM-9083](https://issues.apache.org/jira/browse/BEAM-9083) "PR9677 breaks ValidatesRunnerTest of open source runners"

**Comment by iemejia** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
R: @lukecwik, @kennknowles

Now all checks, except the ones in BEAM-9083 (no ETA), passed.

**Comment by suztomo** [comment]
@iemejia Thank you.

**Comment by markflyhigh** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Run Spark ValidatesRunner

**Comment by lukecwik** [comment]
#10564 may have addressed the spark issues

**Comment by lukecwik** [comment]
The dataflow VR test failure of `Unknown streaming source: test_stream` is being investigated in https://lists.apache.org/thread.html/r65d981d36d91bb2fd85ddedac3db4256c97f08747c38384166fe5481%40%3Cdev.beam.apache.org%3E

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Run Dataflow ValidatesRunner

**Comment by suztomo** [comment]
The last two builds seem to have triggered but did not start.

**Comment by tvalentyn** [comment]
Run Dataflow ValidatesRunner

**Comment by tvalentyn** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
@tvalentyn  Now it has "Details" button. Thank you.

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by suztomo** [comment]
"Google Cloud Dataflow Runner ValidatesRunner Tests" got stuck at 13:07. With test failures:

```
13:07:40 org.apache.beam.sdk.testing.PAssertTest > testWindowedIsEqualTo FAILED
13:07:40     java.lang.RuntimeException at PAssertTest.java:347
13:08:04 
13:08:04 org.apache.beam.sdk.testing.PAssertTest > testContainsInAnyOrderFalse FAILED
13:08:04     java.lang.AssertionError at PAssertTest.java:468
```
https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_PR/189/console

**Comment by suztomo** [comment]
Java PreCommit failed:

```
org.apache.samza.SamzaException: Error opening RocksDB store beamStore at location /tmp/store1
```
https://builds.apache.org/job/beam_PreCommit_Java_Phrase/lastCompletedBuild/

**Comment by suztomo** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Run Dataflow ValidatesRunner

**Comment by suztomo** [comment]
Run Dataflow ValidatesRunner

**Comment by suztomo** [comment]
`Dataflow quota error for jobs-per-project quota. Project apache-beam-testing is running 303 jobs.`
https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_PR/190/testReport/junit/org.apache.beam.sdk/PipelineTest/testTupleProjectionTransform/

**Comment by aaltay** [comment]
Run Dataflow ValidatesRunner

**Comment by suztomo** [comment]
Thank you, Ahmet.

**Comment by suztomo** [comment]
21 successful checks!

**Comment by kennknowles** [comment]
We've seen this change break some of our own internal use of Beam. It may be that it is not a breaking change except if a user has things pinned in particular ways. But it is common for minor version bumps in libs like these to actually be breaking changes. Be on the lookout for Beam users for whom this is a breaking change; we have not yet determined the fix.

**Comment by lukecwik** [comment]
Specifically, the internal usage was pinning tcnative which was causing:
java.lang.NoClassDefFoundError: io/netty/internal/tcnative/SSLPrivateKeyMethod at io.netty.handler.ssl.SslContext.newClientContextInternal(SslContext.java:824) at

**Comment by suztomo** [comment]
@kennknowles @lukecwik Thank you for updates. Glad to hear you resolved the problem.

**Pull request merged**